x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/TimeSeriesLifecycleActionsIT.java

```diff
@@ -71,6 +70,22 @@ public void testAllocate() throws Exception {
         ensureGreen(index);
     }
 
+    public void testAllocateActionOnlyReplicas() throws Exception {
+        int numShards = randomFrom(1, 5);
+        int numReplicas = randomFrom(0, 1);
+        int finalNumReplicas = (numReplicas + 1) % 2;
+        createIndexWithSettings(index, Settings.builder().put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, numShards)
+            .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, numReplicas));
+        AllocateAction allocateAction = new AllocateAction(finalNumReplicas, null, null, null);
+        createNewSingletonPolicy(randomFrom("warm", "cold"), allocateAction);
+        updatePolicy(index, policy);
+        assertBusy(() -> {
+            Map<String, Object> settings = getOnlyIndexSettings(index);
+            assertThat(getStepKey(settings), equalTo(TerminalPolicyStep.KEY));
+            assertThat(settings.get(IndexMetaData.INDEX_NUMBER_OF_REPLICAS_SETTING.getKey()), equalTo(String.valueOf(finalNumReplicas)));
+        });
+    }
```

Comment:
Could we also add a test that tests the bug I found (so where we change both replicas and allocate at the same times on a configuration where the resulting nodes would not be enough to allocate the original number of replicas)?